app/logic/spreadsheet.py

```diff
@@ -0,0 +1,515 @@
+
+
+def volunteersMajors():
+    # Majors represented in volunteering
+    majorQuery = (User.select(User.major, fn.COUNT(fn.DISTINCT(EventParticipant.user_id)).alias('count'))
+                      .join(EventParticipant, on=(User.username == EventParticipant.user_id))
+                      .group_by(User.major))
+    major_query_dict = {row.major: float(row.count) for row in majorQuery}
+
+    return major_query_dict
+
+def classLevelsInVolunteering():
+    # Volunteering numbers by class year
+    classLevelQuery = (User.select(User.classLevel, fn.COUNT(fn.DISTINCT(EventParticipant.user_id)).alias('classCount'))
+                           .join(EventParticipant, on=(User.username == EventParticipant.user_id))
+                           .group_by(User.classLevel))
+    class_level_query_dict = {row.classLevel: float(row.classCount) for row in classLevelQuery}
+
+    return class_level_query_dict
+
+
+def repeatVolunteersPerProgram():
+    # Get people who participated in events more than once (individual program)
+    repeatPerProgramQuery = (EventParticipant.select(EventParticipant.user_id,(ProgramEvent.program_id).alias('program_id'),Program.programName.alias("programName"),fn.Count(EventParticipant.event_id).alias('event_count'))
+                                             .join(ProgramEvent, on=(EventParticipant.event_id == ProgramEvent.event_id))
+                                             .join(Program, on=(ProgramEvent.program_id == Program.id))
+                                             .group_by(EventParticipant.user_id, ProgramEvent.program_id)
+                                             .having(fn.Count(EventParticipant.event_id) > 1))
+    
+    return repeatPerProgramQuery
+
+
+def repeatVolunteersAllPrograms():
+    # Get people who participated in events more than once (all programs)
+    repeatAllProgramQuery = (EventParticipant.select(EventParticipant.user_id, fn.COUNT(EventParticipant.user_id).alias('count'))
+                                             .group_by(EventParticipant.user_id)
+                                             .having(fn.COUNT(EventParticipant.user_id) > 1))
+    return repeatAllProgramQuery
+
+def retentionRate():
+    #fall participation
+    fallParticipationQuery=(ProgramEvent.select(ProgramEvent.program_id, EventParticipant.user_id.alias('participants'), Program.programName.alias("progName"))
+                                        .join(EventParticipant, on=(ProgramEvent.event == EventParticipant.event))
+                                        .join(Program, on=(Program.id == ProgramEvent.program_id))
+                                        .join(Event, on=(EventParticipant.event_id == Event.id))
+                                        .join(Term, on=(Event.term_id == Term.id) )
+                                        .where(Term.description == "Fall 2022"))
+                                  
+
+    fallParticipationDict = {}
+    for result in fallParticipationQuery.dicts():
```

Comment:
Since the fall and spring are becoming one query then we only need one of these. Do it with a defaultdict instead of a normal dict to simplify the for loop.